sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -256,7 +256,10 @@ case class UnresolvedFunction(
   override lazy val resolved = false
 
   override def prettyName: String = name.unquotedString
-  override def toString: String = s"'$name(${children.mkString(", ")})"
+  override def toString: String = {
+    val distinct = if (isDistinct) "distinct " else ""
+    s"'$name($distinct${children.mkString(", ")})"
```

Comment:
We need to handle `filter`, too?

Reply:
filter as a child already exists.

Reply:
Ah, I see.

Reply:
The none is introduced by `unresolvedalias`, shall we reformat it in this pr ?

Reply:
Seems we can check none in `Expression.flatArguments` to avoid these case.

```
  protected def flatArguments: Iterator[Any] = stringArgs.flatMap {
    case t: Iterable[_] => t
    case single => single :: Nil
    case None => Nil
  }
```

Reply:
Oh, ok. Yea, can you?

Reply:
Does this like ok ? please tell me if need to do anything else.

Reply:
Could you add tests somewhere?

Reply:
How about add test in `explain.sql` ? like this
```
-- distinct func
EXPLAIN EXTENDED
  SELECT sum(distinct val)
  FROM explain_temp1
```

Reply:
sgtm